lib/mix/lib/mix/task.ex

```diff
@@ -275,6 +277,13 @@ defmodule Mix.Task do
     end
   end
 
+  defp output_task_debug_info(task, args, proj) do
+    proj_str = if proj, do: " (inside #{inspect proj})", else: ""
+    Mix.shell.info("** Running mix #{task_to_string(task, args)}" <> proj_str)
```

Comment:
I moved `project_to_string` to its own function, but I would rather keep 
`output_task_debug_info` as a function to reduce the noise to `run_task/3`.